metaseq/checkpoint_utils.py

```diff
@@ -105,77 +84,34 @@ def is_better(a, b):
     checkpoints = [
         os.path.join(cfg.save_dir, fn) for fn, cond in checkpoint_conds.items() if cond
     ]
-    if len(checkpoints) > 0:
-        if PathManager.islink(checkpoints[0]):
-            PathManager.rm(checkpoints[0])
```

Comment:
Hm.. I think we might still want to keep this cleanup logic - what was the reason for removal?

Reply:
looks like w're mostly getting rid of the "if len(checkpoints) > 0" line and unindenting the rest. This seems congruent with L78 hardcoding that the size of checkpoint_conds will always be >= 1

Reply:
Yes, Stephen is right. It is only removing  "if len(checkpoints) > 0" and unindenting the rest. Since we removed all no_save* flags, there is no way for "save_checkpoint" to be called without actually saving at least one checkpoint.